app/src/main/java/com/example/lunark/SplashScreenActivity.java

```diff
@@ -1,21 +1,46 @@
 package com.example.lunark;
 
 import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
 import android.content.Intent;
 import android.os.Bundle;
+import android.provider.Settings;
+import android.widget.Toast;
 
-import androidx.appcompat.app.AppCompatActivity;
+import com.example.lunark.tools.CheckConnectionTools;
+import com.google.android.material.snackbar.Snackbar;
 
 import java.util.Timer;
 import java.util.TimerTask;
 
 public class SplashScreenActivity extends Activity {
+    private static final int SPLASH_SCREEN_TIMEOUT = 5000;
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.splash_screen);
 
-        int SPLASH_SCREEN_TIMEOUT = 5000;
+        if(!isConnected()) {
+            Toast.makeText(SplashScreenActivity.this, "Not connected to the Internet", Toast.LENGTH_SHORT).show();
+            showConnectionDialog();
+        } else {
+            openLoginScreen();
+        }
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if(!isConnected()){
+            Toast.makeText(this, "Failed to connect to the internet", Toast.LENGTH_SHORT).show();
```

Comment:
I think "Failed to connect to the internet" should go in strings.xml :thinking:

Reply:
Yes it should. Good call.